apps/dashboard/app/views/files/index.html.erb

```diff
@@ -404,8 +404,8 @@ $('#download-btn').on("click", () => {
   let selection = table.rows({ selected: true }).data();
   let file = selection[0]
 
-  if (selection.length != 1){
-    Swal.fire('Select only 1 file to download', 'You have selected none or multiple rows', 'error')
+  if (selection.length != 1) {
+    selection.toArray().forEach( (f) =>  downloadFile(f) );
```

Comment:
This silently fails when you download a couple files that work and a directory that doesn't. There's no notification that the directory failed. We could defer it because it' may be preferable to downloading things one at a time, but I wonder how difficult the change would be to check if you can download first (and raise an alert) then download if you can.

Reply:
The solution here is for each selected file we still need to do this:

```js
selection.toArray().forEach( (f) =>  {
  if(file.type == 'd') {
    downloadDirectory(file)
  }
  else {
    downloadFile(file)
  }
}
```

Also there isn't any need to check for `  if (selection.length != 1) {` anymore. Just iterate over all the selected files and call downloadDirectory or downloadFile